File: ELEC881_Tabnine/src/No Description Prompt/Python/q16-combination-sum-iv.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        dp = [0] * (target + 1)
        dp[0] = 1
        for num in nums:
            for i in range(num, target + 1):
                dp[i] += dp[i - num]
        return dp[target]
        

File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q16-combination-sum-iv.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var combinationSum4 = function(nums, target) {
    // dp[i] is the number of combinations that add up to i
    let dp = new Array(target + 1).fill(0);
    dp[0] = 1;
    for (let i = 1; i <= target; i++) {
        for (let num of nums) {
        if (i >= num) {
            dp[i] += dp[i - num];
        }
        }
    }
    return dp[target];
    
};